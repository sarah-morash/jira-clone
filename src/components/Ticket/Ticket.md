<!-- prettier- ignore -->
Assigned
```
<Ticket
  id="KGSM-123"
  description="As a user I want to be able to see my components in the styleguide."
  assignedTo="K.G. OHearon"
  estimatedTime={20}
  loggedTime={10}
/>
```

Unassigned
```
<Ticket
  id="KGSM-123"
  description="Some Description"
  estimatedTime={20}
  loggedTime={10}
/>
```
