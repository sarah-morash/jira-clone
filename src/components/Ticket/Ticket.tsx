import { Card } from "@material-ui/core";
import * as React from "react";
import styled from "react-emotion";

export interface ITicket {
  /**
   * The Ticket ID
   */
  id: string;

  /**
   * The description of
   * the ticket
   */
  description: string;

  /**
   * The name of the person
   * assigned to a ticket
   */
  assignedTo?: string;

  /**
   * The original time estimate
   * of the ticket
   */
  estimatedTime: number;

  /**
   * The time logged against
   * this ticket
   */
  loggedTime: number;
}

const CardContentWrapper = styled.div`
  padding: 10px;
`;

const CardIdWrapper = styled.div`
  padding-bottom: 5px;
`;

const CardDescriptionWrapper = styled.div`
  padding: 5px 0;
`;

const CardMetaDataWrapper = styled.div`
  padding-top: 5px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
export default (props: ITicket) => (
  <Card>
    <CardContentWrapper>
      <CardIdWrapper>{props.id}</CardIdWrapper>
      <CardDescriptionWrapper> {props.description} </CardDescriptionWrapper>
      <CardMetaDataWrapper>
        <div> {props.assignedTo ? props.assignedTo : "Unassigned"} </div>
        <div> {props.estimatedTime - props.loggedTime} </div>
      </CardMetaDataWrapper>
    </CardContentWrapper>
  </Card>
);
