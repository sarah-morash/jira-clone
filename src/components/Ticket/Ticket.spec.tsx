import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import Ticket from "./Ticket";

describe("Ticket Component", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Ticket
        id="KGSM-123"
        description="Some Description"
        assignedTo="K.G. OHearon"
        estimatedTime={20}
        loggedTime={10}
      />
    );
  });

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  /**
   * Ticket ID
   */
  it("Should clearly display a ticket number", () => {
    expect(wrapper.html()).toMatch("KGSM-123");
  });

  /**
   * Description
   */
  it("Should clearly display the ticket description", () => {
    expect(wrapper.html()).toMatch("Some Description");
  });

  it("Should trim the description to a maximum of 50 characters", () => {
    const longDescription = new Array(56).join("A");
    expect(wrapper.html()).toMatch(longDescription.slice(55, 1));
  });

  /**
   * Assinged To
   */
  it("Should display the name of the person to whom the ticket is assigned, should it be provided", () => {
    expect(wrapper.html()).toMatch("K.G. OHearon");
  });

  it("Should display the text 'Unassigned' should the ticket currently be unassigned", () => {
    wrapper.setProps({ assignedTo: undefined });
    expect(wrapper.html()).toMatch("Unassigned");
  });

  /**
   * Time Remaining
   */
  it("Should display the time remaining based on the original estimate prop and time logged prop", () => {
    wrapper.setProps({ timeEstimate: 20, timeLogged: 10 });
    expect(wrapper.html()).toMatch("10");

    wrapper.setProps({ timeEstimate: 10, timeLogged: 10 });
    expect(wrapper.html()).toMatch("0");
  });
});
