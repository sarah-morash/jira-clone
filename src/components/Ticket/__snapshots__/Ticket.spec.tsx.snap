// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket Component Should match its snapshot 1`] = `
<WithStyles(Card)>
  <CardContentWrapper>
    <CardIdWrapper>
      KGSM-123
    </CardIdWrapper>
    <CardDescriptionWrapper>
       
      Some Description
       
    </CardDescriptionWrapper>
    <CardMetaDataWrapper>
      <div>
         
        K.G. OHearon
         
      </div>
      <div>
         
        10
         
      </div>
    </CardMetaDataWrapper>
  </CardContentWrapper>
</WithStyles(Card)>
`;
